src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13496,11 +13495,22 @@ void Unit::UpdateSpeed(UnitMoveType mtype, bool forced)
             break;
     }
 
-    // for creature case, we check explicit if mob searched for assistance
-    if (GetTypeId() == TYPEID_UNIT)
+    int32 slowFromHealth = 0;
+    Creature* creature = ToCreature();
+    // ignore pets, player owned vehicles, and mobs immune to snare
+    if (creature
```

Comment:
It should work for bosses too?

Reply:
Is it a remark saying "it should work for bosses too", or is it "but this will slow down bosses"? Bosses should be covered by the `hasMechanicTemplateImmunity(MECHANIC_SNARE)` part. At least it seemed to work when I tested on some?

Reply:
Okay. Not all bosses are immuned to snare effects.

Reply:
Hmm. Then it needs discussion, but are you sure about that?
`SELECT * FROM `creature_template` WHERE `rank` = 3 AND NOT (`mechanic_immune_mask` & 1024) AND `mechanic_immune_mask`!= 0;`
Doesn't seem to return much, and most of those are likely wrong (ie they have 8 388 624 which matches only immune horror and immune fear. I doubt there's that many bosses one can sheep/sap/ban/freeze/stun/snare/slow/grip/charm.

Of course it could be that I have the SQL request wrong. I can make it more explicit if we want and add a check for `isWorldBoss` and/or `isDungeonBoss`

Reply:
Hmm. Then I might be wrong about that. But in case that some bosses will be not immuned to snare, it would be better to use `isDungeonBoss` check.

Reply:
Added now